functions/src/get-reading/purple-air-response.ts

```diff
@@ -113,36 +165,69 @@ function getReading(
     humidity !== undefined && // Can be zero
     timestamp
   ) {
-    return [
-      id,
-      {
-        id: id,
-        name: name,
-        latitude: latitude,
-        longitude: longitude,
-        meanPercentDifference: meanPercentDifference,
-        pm25: pm25,
-        humidity: humidity,
-        timestamp: timestamp,
-      },
-    ];
+    const reading: PurpleAirReading = {
+      id: id,
+      name: name,
+      latitude: latitude,
+      longitude: longitude,
+      meanPercentDifference: meanPercentDifference,
+      pm25: pm25,
+      humidity: humidity,
+      timestamp: timestamp,
+    };
+
+    /**
+     * Sometimes PurpleAir will give a non-zero confidence value when a sensor
+     * has a channel downgraded. This is not the behavior we want because we
+     * need both channels' readings to be able to determine if we want to use
+     * that reading in our AQI calculation since we want to be able to know how
+     * confident we can be that a sensor reading is true and not an outlier.
+     * Thus, if either channel of a sensor is downgraded, we set the
+     * meanPercentDifference to the maximum value of 2.
+     */
+    if (
+      sensorErrors[SensorReadingErrors.ChannelADowngraded] ||
+      sensorErrors[SensorReadingErrors.ChannelADowngraded]
```

Comment:
copy paste error, want `ChannelBDowngraded`

Reply:
oops thank you